crates/forge/tests/it/zk/factory.rs

```diff
@@ -0,0 +1,75 @@
+//! Forge tests for zksync factory contracts.
+
+use super::test_zk;
+
+test_zk!(can_deploy_in_method, "testClassicFactory|testNestedFactory", "ZkFactoryTest");
+
+test_zk!(
+    can_deploy_in_constructor,
+    "testConstructorFactory|testNestedConstructorFactory",
+    "ZkFactoryTest"
+);
+
+test_zk!(can_use_predeployed_factory, "testUser.*", "ZkFactoryTest");
+
+foundry_test_utils::forgetest_async!(script_zk_can_deploy_in_method, |prj, cmd| {
+    setup_factory_prj(&mut prj);
+    execute_factory_script(prj.root(), &mut cmd, "ZkClassicFactoryScript", 2);
+    execute_factory_script(prj.root(), &mut cmd, "ZkNestedFactoryScript", 2);
+});
+
+foundry_test_utils::forgetest_async!(script_zk_can_deploy_in_constructor, |prj, cmd| {
+    setup_factory_prj(&mut prj);
+    execute_factory_script(prj.root(), &mut cmd, "ZkConstructorFactoryScript", 1);
+    execute_factory_script(prj.root(), &mut cmd, "ZkNestedConstructorFactoryScript", 1);
+});
+
+foundry_test_utils::forgetest_async!(script_zk_can_use_predeployed_factory, |prj, cmd| {
+    setup_factory_prj(&mut prj);
+    execute_factory_script(prj.root(), &mut cmd, "ZkUserFactoryScript", 3);
+    execute_factory_script(prj.root(), &mut cmd, "ZkUserConstructorFactoryScript", 2);
+});
+
+fn setup_factory_prj(prj: &mut foundry_test_utils::TestProject) {
+    foundry_test_utils::util::initialize(prj.root());
+    prj.add_source("Factory.sol", include_str!("../../../../../testdata/zk/Factory.sol")).unwrap();
+    prj.add_script("Factory.s.sol", include_str!("../../fixtures/zk/Factory.s.sol")).unwrap();
+}
+
+fn execute_factory_script(
```

Comment:
```suggestion
fn test_factory_script(
```
(Feel free to ignore this, just opinionated nit): `execute` feels like you are just running the script but in reality the method is both executing + testing it is successful + testing it generates the expected transactions

Reply:
or `run_script_test`